source/errors/errors_observer.cpp

```diff
@@ -15,6 +15,8 @@ cmsl::string_view to_string(cmsl::errors::error_type type)
       return "warning";
     case cmsl::errors::error_type::note:
       return "note";
+    default:
```

Comment:
That might be error prone.
Why not simply stick to the rule, that you enumerate over every single error_type?

Reply:
Well, you're probably right. The `default` should never be executed. I'll put a `CMSL_UNREACHABLE()` there. The `return` needs to be there, though, to satisfy `-Werror`.

Reply:
Then why not simply:
```
switch (type) {
    case cmsl::errors::error_type::error:
        return "error";
    case cmsl::errors::error_type::warning:
        return "warning";
    case cmsl::errors::error_type::note:
        return "note";
}
return "<unknown>";
```
?

Reply:
gcc complains about missing default /shrug

Reply:
Additionally, if a `switch` is the only thing in the function, IMO it's better to handle everything in the switch.

Reply:
Ok, my bad, gcc doesn't complain if there is a `return` at the end of a function. But, still, I think that handling everything in the switch is better in this case

Reply:
Isn't that because you have not covered every single enumeration in a switch statement?
i.e.: https://godbolt.org/z/Y4cpIE

or am I missing something in here? (haven't been checking that with actual code, locally).

Reply:
All enumerators are covered there. If the `return` at the end of function is present, everything works like a charm.

Reply:
Ok, then my godbolt is invalid ^^